internal/git/gitsync_processor.go

```diff
@@ -157,8 +185,228 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, rest
+
+func ApplyPatchToResources(patchedResources PatchedResource, client kubernetes.Client) error {
+	for key, afterValue := range patchedResources.After {
+		namespace := strings.Split(key, "/")[0]
+		if beforeValue, ok := patchedResources.Before[key]; ok {
+			if beforeValue != afterValue {
+				err := client.ApplyResource([]byte(afterValue), namespace)
+				if err != nil {
+					return errors.New("failed to apply resource: " + err.Error())
+				}
+			}
+		} else {
+			// Handle newly added resources
+			err := client.ApplyResource([]byte(afterValue), namespace)
+			if err != nil {
+				return errors.New("failed to apply new resource: " + err.Error())
+			}
+		}
+	}
+
+	// Handle deleted resources
+	for key, beforeValue := range patchedResources.Before {
+		resource, err := yamlUnmarshal(beforeValue)
+		if err != nil {
+			return errors.New("failed to unmarshal YAML: " + err.Error())
+		}
+		if _, ok := patchedResources.After[key]; !ok {
+			err := client.DeleteResource(resource.Kind, resource.Metadata.Name, resource.Metadata.Namespace, metav1.DeleteOptions{})
+			if err != nil {
+				return errors.New("failed to delete resource: " + err.Error())
+			}
+		}
+	}
+	return nil
+}
+
+// this check for file changes in the repo by comparing the old commit  hash with new commit hash and returns the recent hash with PatchedResources map
+
+func CheckForRepoUpdates(ctx context.Context, r *git.Repository, repo *v1.RepositoryPath, status *v1.GitSyncStatus, defaultNameSpace string) (PatchedResource, string, error) {
+	var patchedResources PatchedResource
+	var recentHash string
+	logger := logging.FromContext(ctx)
+	if err := fetchUpdates(r); err != nil {
+		logger.Errorw("error checking for updates in the github repo", "err", err, "repo", repo.RepoUrl)
+		return patchedResources, recentHash, err
+	}
+	remoteRef, err := getLatestCommitHash(r, repo.TargetRevision)
+	if err != nil {
+		logger.Errorw("failed to get latest commits in the github repo", "err", err, "repo", repo.RepoUrl)
+		return patchedResources, recentHash, err
+	}
+	lastCommitStatus := status.CommitStatus[repo.Name]
+	if remoteRef.String() != lastCommitStatus.Hash {
+		recentHash = remoteRef.String()
+		lastTreeForThePath, err := getCommitTreeAtPath(r, repo.Path, plumbing.NewHash(lastCommitStatus.Hash))
+		if err != nil {
+			logger.Errorw("failed to  get last commit", "err", err, "repo", repo.RepoUrl)
+			return patchedResources, recentHash, err
+		}
+
+		recentTreeForThePath, err := getCommitTreeAtPath(r, repo.Path, *remoteRef)
+
+		if err != nil {
+			logger.Errorw("failed to  recent commit", "err", err, "repo", repo.RepoUrl)
+			return patchedResources, recentHash, err
+		}
+
+		patch, err := lastTreeForThePath.Patch(recentTreeForThePath)
+		if err != nil {
+			logger.Errorw("failed to patch commit", "err", err, "repo", repo.RepoUrl)
+			return patchedResources, recentHash, err
+		}
```

Comment:
```suggestion
                 // get the diff between the previous and current 
		patch, err := lastTreeForThePath.Patch(recentTreeForThePath)
		if err != nil {
			logger.Errorw("failed to take diff of commit", "err", err, "repo", repo.RepoUrl)
			return patchedResources, recentHash, err
		}
```

Reply:
(I just find the word "patch" confusing because to me it seems like we're making a change but really just taking a diff)